install-deps.sh

```diff
@@ -19,12 +19,43 @@ if test $(id -u) != 0 ; then
 fi
 export LC_ALL=C # the following is vulnerable to i18n
 
+function ensure_decent_gcc_on_deb {
+    # point gcc to the one offered by distro if the used one is different
+    local old=$(readlink -e /usr/bin/x86_64-linux-gnu-gcc | cut -d'-' -f2)
+    local new=$1
+    if dpkg --compare-versions $old eq $new; then
+	    return
+    fi
+
```

Comment:
maybe display a big fat warning here? Something like: "The GCC on this system is not the version that ships with the distro. Reverting!"

Reply:
updated and repushed.

Reply:
Just out of curiosity - why display the warning only on interactive shells?

Reply:
i think, it only makes sense to show warning to human, otherwise, it's likely to be run by our jenkins builder. in that case, it's us who will be reading these warnings. and as far as i am concerned, i don't really want to read them.